aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -967,6 +967,22 @@ Tensor tensor_backend(ArrayRef<T> values, const TensorOptions& options) {
   return cpu_tensor.to(options.device());
 }
 
+template <typename T>
+Tensor tensor_complex_cpu(ArrayRef<T> values, const TensorOptions& options) {
```

Comment:
why do we need this? `tensor_cpu` above is already being dispatched for complex and this function looks to be doing the exact thing as above

Reply:
This is for converting **from complex to complex**.
`tensor_cpu` supports converting **from real to any dtype**.

We can not convert **from complex to any dtype**, it will fail with compile error like "can not convert c10::complex<float> to int".

Reply:
isn't that because in the `TENSOR` macro we don't have complex dtypes included?
```
AT_FORALL_SCALAR_TYPES_AND3(Bool, Half, BFloat16, TENSOR)
```

Reply:
I don't think so, I believe it is in:
```C++
std::copy(values.begin(), values.end(), result.template data_ptr<scalar_t>());
```
where `values` is complex type, and `result.template data_ptr<scalar_t>()` is int pointer. And internally C++ does
```C++
*ptr = value;
```
but this is wrong because we can not convert complex to int.